### PR TITLE
feat(presets): ai create an image

### PR DESCRIPTION
**Pull request by fundon**
feat(presets): ai create an image

* [x] text to image
* [x] image to image

### CODE REVIEW

**Comment by fundon** [comment]
* **#6746** <a href="https://app.graphite.dev/github/pr/toeverything/blocksuite/6746?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @fundon and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: packages/presets/src/ai/actions/edgeless-response.ts [hunk 1]
```diff
@@ -175,6 +176,49 @@ export const responses: {
...
+
+    const isDataUrl = data.startsWith('data:image/');
+    const filename = 'image';
+    let imgPromise;
+
+    if (isDataUrl) {
+      imgPromise = Promise.resolve(dataURLtoFile(data, filename));
+    } else if (URL.canParse(data)) {
+      // @TODO: need a proxy
+      imgPromise = fetchImageToFile(data, filename);
```
Comment (fundon) [inline packages/presets/src/ai/actions/edgeless-response.ts:200]: @darkskygit need a api to download images

**Review by Flrande — APPROVED** [review]

File: packages/presets/src/ai/utils/event-source.ts [hunk 2]
```diff
@@ -5,10 +5,12 @@ function delay(ms: number) {
 // do we really need to export this?
 export function toTextStream(
   eventSource: EventSource,
-  options?: {
+  options: {
     timeout?: number;
-  }
+    type?: string;
```
Comment (pengx17) [inline packages/presets/src/ai/utils/event-source.ts:10]: why adding type here? I think the only event type not being listened to is `open`. There is no need to expose this to the outside

Comment (fundon) [reply packages/presets/src/ai/utils/event-source.ts:10]: Need to listen `attachment` event.

https://github.com/toeverything/AFFiNE/pull/6538/files#diff-e8a9e3cb567d0cc9ab5e33a5ad6cba230be641ad804f7b8440c852fcdbb98333R175

**Review by Flrande — APPROVED** [review]

**Review by pengx17 — APPROVED** [review]

**Pull request merged**